compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5290,7 +5290,7 @@ private BSymbol getLangLibMethod(BLangInvocation iExpr, BType bType) {
     }
 
     private void checkInvocationParamAndReturnType(BLangInvocation iExpr) {
-        BType actualType = checkInvocationParam(iExpr);
+        BType actualType = iExpr.originalType = checkInvocationParam(iExpr);
```

Comment:
While this change fixes the issue, wonder if we can remove the check for `INVOCATION` at https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L7770 altogether. 

IIRC, previously error/nil lifting access was allowed on invocations, but that is not allowed anymore. So we may be able to remove that condition. Shall we check if this is possible?

Reply:
Changed

Reply:
We don't need this change anymore, right?

Reply:
Yes, removed